Lib9c/Action/ClaimItems.cs

```diff
@@ -0,0 +1,114 @@
+        private const string ActionTypeText = "claim_items";
+
+        public IEnumerable<Address> AvatarAddresses { get; private set; }
+        public IEnumerable<FungibleAssetValue> Amounts { get; private set; }
+
+        public ClaimItems() {}
+
+        public ClaimItems(IEnumerable<Address> avatarAddresses, IEnumerable<FungibleAssetValue> amounts)
+        {
+            AvatarAddresses = avatarAddresses;
+            Amounts = amounts;
+        }
+
+        protected override IImmutableDictionary<string, IValue> PlainValueInternal =>
+            ImmutableDictionary<string, IValue>.Empty
+                .Add(AvatarAddressKey, new List(AvatarAddresses.Select(x => x.Serialize())))
+                .Add(AmountKey, new List(Amounts.Select(x => x.Serialize())));
+
+        protected override void LoadPlainValueInternal(
+            IImmutableDictionary<string, IValue> plainValue)
+        {
+            AvatarAddresses = ((List)plainValue[AvatarAddressKey]).Select(x => x.ToAddress());
+            Amounts = ((List)plainValue[AmountKey]).Select(x => x.ToFungibleAssetValue());
+        }
+
+        public override IAccount Execute(IActionContext context)
+        {
+            context.UseGas(1);
+
+            var states = context.PreviousState;
+            var itemSheet = states.GetSheets(containItemSheet: true).GetItemSheet();
+
+            var inventories = AvatarAddresses.Select(address =>
+            {
+                var addressHex = GetSignerAndOtherAddressesHex(context, address);
+                var inventory = states.GetInventory(address.Derive(LegacyInventoryKey));
+                if (inventory is null)
+                {
+                    throw new FailedLoadStateException(ActionTypeText, addressHex, typeof(AvatarState), address);
+                }
+
+                return inventory;
+            })
+                .ToList();
+
+            foreach (var fungibleAssetValue in Amounts)
+            {
+                var ticker = fungibleAssetValue.Currency.Ticker;
+                if (!ticker.StartsWith("IT_") ||
+                    !int.TryParse(ticker.Replace("IT_", string.Empty), out var itemId))
+                {
+                    throw new ArgumentException($"Format of Amount currency's ticker is invalid");
+                }
+
+                var decimalPlaces = fungibleAssetValue.Currency.DecimalPlaces;
+                if (decimalPlaces != 0)
+                {
+                    throw new ArgumentException(
+                        "DecimalPlaces of fungibleAssetValue for claimItems are not 0");
+                }
+
+                var balance = states.GetBalance(context.Signer, fungibleAssetValue.Currency);
+                if (balance < fungibleAssetValue)
+                {
+                    throw new NotEnoughFungibleAssetValueException(
+                        context.Signer.ToHex(),
+                        fungibleAssetValue.RawValue,
+                        balance);
+                }
```

Comment:
How about this? We just burn the asset here.
```suggestion
                states = states.BurnAsset(context, context.Signer, fungibleAssetValue * inventories.Count);
```